cpssEnabler/linuxNoKernelModule/drivers/intDriver.c

```diff
@@ -301,14 +359,25 @@ static ssize_t mvIntDrv_write(struct file *f, const char *buf, size_t siz, loff_
 						     (unsigned)cmdBuf[5]));
 	if (pdev) {
 		int rc;
+
+		pr_info("%s: Got request to enable MSI for %x:%x:%x\n",
+				__func__, (((cmdBuf[2]<<8)&0xff00)|(cmdBuf[1]&0xff)),
+					   (unsigned)cmdBuf[3],
+					   PCI_DEVFN((unsigned)cmdBuf[4],
+						     (unsigned)cmdBuf[5]));
+
+		if (mvintdrv_add_pci_dev_to_ar(pdev)) {
+			printk("%s: Cannot reg pdev %p!\n", __func__, pdev);
+			} else pr_info("%s: Queueing pci device for driver cleanup MSI disablement...\n", __func__);
+
 		if (pci_dev_msi_enabled(pdev)) {
-			pci_dev_put(pdev);
+			pr_info("%s: PCIe MSI already enabled.\n", __func__);
 			return 0;
 		}
 		rc = pci_enable_msi(pdev);
```

Comment:
FWIW, this is marked as deprecated in kernel source code (drivers/pci/msi.c)

Reply:
Will be forwarded to CPSS team for future resolution